src/script/myRenderer.js

```diff
@@ -60,14 +72,31 @@ export class Renderer {
         });
     } 
 
-        /* renderByTag () {
+        renderType (data) {
             // adds eventlistener to ByType checkboxes
+            const filterer = new Filter(data);
```

Comment:
This line "tightly couples" the renderer class with the `Filter` class. This means that you can't use different filter classes with the renderer, even if they play the same role and act the same way, because it's hard coded here what class to use.

If the filter was passed into this class via the constructor, this class wouldn't have to understand completely what a filter is, only how to use it.